tests/test_executable.py

```diff
@@ -137,3 +140,55 @@ def test_executable_not_privileged(
 
     collapsed = parse_one_collapsed(Path(output_directory / "last_profile.col").read_text())
     assert_collapsed(collapsed)
+
+
+@pytest.mark.parametrize(
+    "runtime,profiler_type",
+    [
+        ("python", "py-spy"),
+    ],
+)
+@pytest.mark.parametrize("in_container", [True])
+@pytest.mark.parametrize("application_docker_mount", [True])
+@pytest.mark.parametrize("application_docker_capabilities", ["SYS_PTRACE"])
+def test_killing_spawned_processes(
+    gprofiler_exe: Path,
+    application_docker_container: Container,
+    runtime_specific_args: List[str],
+    output_directory: Path,
+    profiler_flags: List[str],
+    application_docker_mount: bool,
+) -> None:
+    """Tests if killing gprofiler with -9 results in killing py-spy"""
```

Comment:
I'd give a link to the issue https://github.com/Granulate/gprofiler/issues/674 here which explains a bit more about the reasoning.